ui/src/app/pages/analysis/util.ts

```diff
@@ -2,24 +2,47 @@ import { FileDetail } from 'generated/fetch';
 
 import { notebooksApi } from 'app/services/swagger-fetch-clients';
 
-const notebookExtension = '.ipynb';
+const jupyterNotebookExtension = '.ipynb';
+const rstudioNotebookExtension = '.Rmd';
 
 export function dropNotebookFileSuffix(filename: string) {
-  if (filename?.endsWith(notebookExtension)) {
-    return filename.substring(0, filename.length - notebookExtension.length);
+  if (filename?.endsWith(jupyterNotebookExtension)) {
+    return filename.substring(
+      0,
+      filename.length - jupyterNotebookExtension.length
+    );
   }
 
   return filename;
 }
 
-export function appendNotebookFileSuffix(filename: string) {
-  if (filename && !filename.endsWith(notebookExtension)) {
-    return filename + notebookExtension;
+export function appendJupyterNotebookFileSuffix(filename: string) {
+  if (filename && !filename.endsWith(jupyterNotebookExtension)) {
+    return filename + jupyterNotebookExtension;
   }
 
   return filename;
 }
 
+export function appendRstudioNotebookFileSuffix(filename: string) {
+  if (filename && !filename.endsWith(rstudioNotebookExtension)) {
+    return filename + rstudioNotebookExtension;
+  }
+
+  return filename;
+}
+
+export function appendNotebookFileSuffixByOldName(
+  filename: string,
+  oldFileName: string
+) {
+  if (oldFileName.endsWith(jupyterNotebookExtension)) {
```

Comment:
I'd prefer that each check is explicit, with some default behavior (don't add a suffix, maybe?)

Something like:

```
cond(
  [oldFileName.endsWith(jupyterNotebookExtension), () => appendJupyterNotebookFileSuffix(filename)],
  [oldFileName.endsWith(rstudioNotebookExtension), () => appendRStudioNotebookFileSuffix(filename)],
  () => filename
);
```